File: models/static/scripts/signup.js
$(function () {
	$('.signupbutton').on('click', function () {
		let firstname = $('.form input[name="firstname"]').val()
		let lastname = $('.form input[name="lastname"]').val()
		let email = $('.form input[name="email"]').val()
		let password = $('.form input[name="password"]').val()
		const data = {"first_name": firstname, "last_name": lastname, "email": email, "password": password};

		$.ajax({
			url: "http://0.0.0.0:5001/api/v1/signup",
			data: JSON.stringify(data),
			contentType: "application/json",
			type: "POST",
			success: function (response) {
				if (response.status === "OK") {
					window.location = 'http://0.0.0.0:5000/login';
				} else if (response.status === "emailerror") {
					$('p.error').html("<h3>Email already in use</h3>");
				}

			},
			error: function (response) {
				alert(response.status);
			}
		});



	});
});

File: models/views/signup.py
from models.views import app_views
from flask import render_template, flash, url_for, redirect, request
import models
from models.forms.signup import SignupForm

@app_views.route('/signup', methods=['GET', 'POST'], strict_slashes=True)
def signup():
    if request.method == 'POST':
        form = request.form
        name = form.get('full_name')
        email = form.get('email')
        password1 = form.get('password1')
        password2 = form.get('password2')
        
        try:
            name = name.split()
            first_name = name[0]
            last_name = name[1]

        except IndexError:
            flash("Please Enter full name", category="warning")
            return redirect(url_for('app_views.signup'))
        #if password1 != password2:
        #    flash("password fields must match", category="warning")
        #    return redirect(url_for('app_views.signup'))

        data = {'first_name': first_name, 'last_name': last_name,
                'email': email, 'password': password1}
        from models.user import User
        user = models.storage.get_by_email(email)
        if not user:
            user = User(**data)
            user.save()
            flash("Account created successfully! Please login", category='success')
            return redirect(url_for('app_views.login'))
        flash("Email already in use please try a different email", category="warning")
    return render_template('newsignup.html')
